prometheus_alerting_consistency.md

```diff
@@ -0,0 +1,104 @@
+# Prometheus Alerting Consistency
+
+## Summary
+
+This guide is originally based on the [Alerting Consistency][1] proposal for OpenShift alerts.
+
+Clear and actionable alerts are a key component of a smooth operational experience. If an alert doesn't have a proven documented procedure to follow when it fires, it can lead to an increase in mean time to repair (MTTR).
+Ensuring you have clear and concise guidelines for engineers and SRE creating new alerts will result in a better experience for end users.
+
+## Recommended Reading
+
+A list of references on good alerting practices:
+
+* [Google SRE Book - Monitoring Distributed Systems][2]
+* [Prometheus Alerting Documentation][3]
+* [Alerting for Distributed Systems][4]
+
+## Style Guide
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119).
+
+* Alert names MUST be CamelCase, e.g.: `BackendServiceStuck`
+* Alert names SHOULD be prefixed with a component, e.g.: `ZookeeperPersistentVolumeFillingUp`, or an SLO name if the alert is a [Multi-Window, Multi-Burn Rate alert](https://sre.google/workbook/alerting-on-slos/#6-multiwindow-multi-burn-rate-alerts)
+  * There may be exceptions for some broadly scoped alerts, e.g.: `TargetDown`
+* Alerts MUST include a `severity` label indicating the alert's urgency.
+  * Valid severities are: `critical` or `warning` — see below for
+    guidelines on writing alerts of each severity.
+* Alerts MUST include `summary` and `description` annotations.
+  * Think of `summary` as the first line of a commit message, or an email
+    subject line.  It should be brief but informative.  The `description` is the
+    longer, more detailed explanation of the alert.
+* Where applicable, alerts SHOULD include a `namespace` label indicating the source of the alert.
+  * Many alerts will include this by virtue of the fact that their PromQL
+    expressions result in a namespace label.  Others may require a static
+    namespace label.
+* Alerts SHOULD include a cluster identifier label, especially in cases where alerts are defined or routed centrally for multiple services & clusters.
+  * This can hint to an SRE where the affected service is located. It can also be useful for aggregation & filtering when analysing groups of firing alerts.
+* All alerts SHOULD include an annotation (e.g. `runbook_url`) which directs the responder(s) to the location of the runbook (sometimes called a Standard Operating Procedure or SOP) that is specific to fixing that problem.
```

Comment:
Is this pattern in wide spread use already?

Reply:
> Is this pattern in wide spread use already?

It's based on the guide for Managed Openshift, so I would expect it to be in use there.
Managed Kafka has a version that is very similar (inspired by Managed Openshift).
SRES would use the `runbook_url` for sure, though it is `sop_url` in at least 1 group.
This would cover a lot of teams & services involved with prometheus based alerts.